tests/test_replay.py

```diff
@@ -22,7 +22,7 @@ def ctx():
     execer = Execer(xonsh_ctx=ctx)
     builtins.__xonsh__.shell = Shell(execer=execer, ctx=ctx, shell_type="none")
     yield
-    del builtins.__xonsh__.shell
+    builtins.__xonsh__ = None
```

Comment:
```suggestion
    builtins.__xonsh__.shell = None
```

The flake failures are unrelated but I think this needs to be changed.